src/zip2john.c

```diff
@@ -779,7 +779,8 @@ static int process_legacy(zip_file *zfp, zip_ptr *p)
 				// OLD winzip (I think 8.01 or before), is also supposed to be 2 byte.
 				// old v1 pkzip (the DOS builds) are 2 byte checksums.
 			{
-				zfp->check_bytes = 2;
+				if (allow_2_byte_checksum) // See #4541
+					zfp->check_bytes = 2;
```

Comment:
Isn't it weird/wrong that we only use 2-byte checksum conditionally even after having checked for very specific `efh_id` values? Don't those known values guarantee we have 2-byte? Confused.

Reply:
> Don't those known values guarantee we have 2-byte

OK, I see that the issue we're trying to fix is `0x7875` doesn't guarantee 2-byte checksum.

I think we should add a comment that unfortunately checksum size seen with that EFH value varies between the two known archivers that use that value.

Reply:
So the main question is whether we want to hurt the speed at all other previously-whitelisted EFH values just because we found this problem for `0x7875` now. I doubt we want to do that, or at least we should have `zip2john` recommend use of `-2` for those. In fact, it could print a more verbose recommendation even for `0x7875`, mentioning the `bsdtar` issue.